src/check-reserve-update-frequency/check-reserve-update-frequency.js

```diff
@@ -0,0 +1,166 @@
+  let reserveTokenArray = await protocolDataProviderContract.getAllReservesTokens();
+
+  // remove tokens that do not have Chainlink price oracles
+  reserveTokenArray = reserveTokenArray.filter(
+    (reserveToken) => TOKENS_WITHOUT_CHAINLINK_ABI.indexOf(reserveToken.tokenAddress) === -1,
+  );
+
+  // get the AAVE price oracle address
+  const priceOracleAddress = await lendingPoolAddressProviderContract.getPriceOracle();
+  const priceOracleContractInstance = new ethers.Contract(
+    priceOracleAddress,
+    priceOracleAbi,
+    jsonRpcProvider,
+  );
+
+  // get the price source addresses
+  const priceSourceAddresses = await Promise.all(reserveTokenArray.map(
+    (reserveToken) => priceOracleContractInstance.getSourceOfAsset(reserveToken.tokenAddress),
+  ));
+
+  // create ethers contracts to run read-only methods from the Chainlink contracts
+  const priceSourceContractInstances = priceSourceAddresses.map(
+    (priceSourceAddress) => new ethers.Contract(
+      priceSourceAddress,
+      chainlinkAggregatorAbi,
+      jsonRpcProvider,
+    ),
+  );
+
+  // create an array of token / address / contract tuples that we will iterate over
+  const tokenAddressContractTuples = reserveTokenArray.map((token, index) => {
+    const address = priceSourceAddresses[index];
+    const contract = priceSourceContractInstances[index];
+    return [token, address, contract];
+  });
+
+  return tokenAddressContractTuples;
+}
+
+function provideHandleBlock(tokensAddressesContractsPromise) {
+  return async function handleBlock(blockEvent) {
+    const findings = [];
+
+    // settle the promise the first time, all subsequent times just get the resolve() value
+    const tokensAddressesContracts = await tokensAddressesContractsPromise;
+
+    // get the timestamp for the current block
+    const blockTimestamp = new BigNumber(blockEvent.block.timestamp);
+
+    // override block number so we get data from the block in question
+    const override = { blockTag: blockEvent.blockNumber };
+
+    // define the promise function to run for each reserve token
+    async function checkOracleAge(tokenAddressContract) {
+      const [reserveToken, priceSourceAddress, priceSourceContract] = tokenAddressContract;
+
+      // get the timestamp from the price source contract
+      let roundData;
+      try {
+        roundData = await priceSourceContract.latestRoundData({ ...override });
+      } catch (error) {
+        return;
+      }
+
+      // the updatedAt value is of type ethers.BigNumber
+      // ethers.BigNumber is not the same as BigNumber from bignumber.js
+      // therefore, we need to convert from ethers.BigNumber to BigNumber
+      const timestamp = new BigNumber(roundData.updatedAt.toString());
+
+      // calculate the difference between the current block timestamp and the last oracle update
+      const oracleAge = blockTimestamp.minus(timestamp);
+
+      if (oracleAge.isGreaterThan(ORACLE_AGE_THRESHOLD_SECONDS)) {
```

Comment:
I personally am a fan of globals being all caps, should we start doing this and maybe refactor old code to match this?

Reply:
I'm importing this variable from `agent-config.json` where it's lowerCamelCase and I'm importing it as all caps.  I support all caps signifying globals.